File: src/agent-resources/agent_resources/cli/command.py
"""CLI for command-add command."""

from typing import Annotated

import typer

from agent_resources.cli.common import get_destination, parse_resource_ref
from agent_resources.exceptions import (
    ClaudeAddError,
    RepoNotFoundError,
    ResourceExistsError,
    ResourceNotFoundError,
)
from agent_resources.fetcher import ResourceType, fetch_resource

app = typer.Typer(
    add_completion=False,
    help="Add Claude Code slash commands from GitHub to your project.",
)


@app.command()
def add(
    command_ref: Annotated[
        str,
        typer.Argument(
            help="Command to add in format: <username>/<command-name>",
            metavar="USERNAME/COMMAND-NAME",
        ),
    ],
    overwrite: Annotated[
        bool,
        typer.Option(
            "--overwrite",
            help="Overwrite existing command if it exists.",
        ),
    ] = False,
    global_install: Annotated[
        bool,
        typer.Option(
            "--global",
            "-g",
            help="Install to ~/.claude/ instead of ./.claude/",
        ),
    ] = False,
) -> None:
    """
    Add a slash command from a GitHub user's agent-resources repository.

    The command will be copied to .claude/commands/<command-name>.md in the
    current directory (or ~/.claude/commands/ with --global).

    Example:
        command-add kasperjunge/commit
        command-add kasperjunge/review-pr --global
    """
    try:
        username, command_name = parse_resource_ref(command_ref)
    except typer.BadParameter as e:
        typer.echo(f"Error: {e}", err=True)
        raise typer.Exit(1)

    dest = get_destination("commands", global_install)
    scope = "user" if global_install else "project"

    typer.echo(f"Fetching command '{command_name}' from {username}/agent-resources...")

    try:
        command_path = fetch_resource(
            username, command_name, dest, ResourceType.COMMAND, overwrite
        )
        typer.echo(f"Added command '{command_name}' to {command_path} ({scope} scope)")
    except RepoNotFoundError as e:
        typer.echo(f"Error: {e}", err=True)
        raise typer.Exit(1)
    except ResourceNotFoundError as e:
        typer.echo(f"Error: {e}", err=True)
        raise typer.Exit(1)
    except ResourceExistsError as e:
        typer.echo(f"Error: {e}", err=True)
        raise typer.Exit(1)
    except ClaudeAddError as e:
        typer.echo(f"Error: {e}", err=True)
        raise typer.Exit(1)


if __name__ == "__main__":
    app()

File: src/agent-resources-npm/src/cli/command.ts
/**
 * CLI for add-command command.
 */

import { Command } from 'commander';
import {
  parseResourceRef,
  getDestination,
  createFetchSpinner,
  printSuccessMessage,
} from './common.js';
import { fetchResource } from '../fetcher.js';
import { ResourceType } from '../types.js';
import {
  ClaudeAddError,
  RepoNotFoundError,
  ResourceExistsError,
  ResourceNotFoundError,
} from '../exceptions.js';

/**
 * Create the add-command CLI command.
 */
export function createCommandCommand(): Command {
  const program = new Command()
    .name('add-command')
    .description('Add Claude Code commands from GitHub to your project.')
    .argument(
      '<command-ref>',
      'Command to add in format: <username>/<command-name>'
    )
    .option('--overwrite', 'Overwrite existing command if it exists.', false)
    .option(
      '-g, --global',
      'Install to ~/.claude/ instead of ./.claude/',
      false
    )
    .action(
      async (
        commandRef: string,
        options: { overwrite: boolean; global: boolean }
      ) => {
        try {
          const [username, commandName] = parseResourceRef(commandRef);
          const dest = getDestination('commands', options.global);

          const spinner = createFetchSpinner();
          spinner.start();

          try {
            await fetchResource(
              username,
              commandName,
              dest,
              ResourceType.COMMAND,
              options.overwrite
            );
            spinner.stop();
            printSuccessMessage('command', commandName, username);
          } catch (error) {
            spinner.stop();
            throw error;
          }
        } catch (error) {
          if (
            error instanceof RepoNotFoundError ||
            error instanceof ResourceNotFoundError ||
            error instanceof ResourceExistsError ||
            error instanceof ClaudeAddError
          ) {
            console.error(`Error: ${error.message}`);
            process.exit(1);
          }
          if (error instanceof Error) {
            console.error(`Error: ${error.message}`);
            process.exit(1);
          }
          throw error;
        }
      }
    );

  return program;
}
